### PR TITLE
Add Plot Twist No Fansub

### LINKED ISSUE #110: Plot twist no fansub
### Source name

Plot Twist No Fansub

### Source link

https://www.plotfansub.com/

### Source language

Español

### Other details

_No response_

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/keiyoushi/extensions-source/) and verified it does not appear in the code base.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by bapeey**
Closes #110 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            .build()
+
+        return POST(url, ajaxHeaders, body)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val result = json.decodeFromString<SearchResultDto>(response.body.string())
+        val unescapedHtml = result.data.unescape()
+        val newResponse = response.newBuilder().body(unescapedHtml.toResponseBody(HTML_MEDIA_TYPE)).build()
+        return super.searchMangaParse(newResponse)
```
Comment (AwkwardPeak7) [inline src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt]: just implement the element mapping instead

File: src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun latestUpdatesFromElement(element: Element): SManga = throw UnsupportedOperationException()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/wp-admin/admin-ajax.php"
+
+        val ajaxHeaders = this.headersBuilder()
+            .add("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
```
Comment (AwkwardPeak7) [inline src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt]: afaik okhttp will automatically set Content-Type header based on the request body

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

**Review by AwkwardPeak7 — APPROVED** [review]

File: src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt [hunk 3]
```diff
@@ -75,25 +74,21 @@ class PlotTwistNoFansub : ParsedHttpSource() {
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
         val url = "$baseUrl/wp-admin/admin-ajax.php"
 
-        val ajaxHeaders = this.headersBuilder()
-            .add("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
-            .add("X-Requested-With", "XMLHttpRequest")
```
Comment (AwkwardPeak7) [inline src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt]: is `XMLHttpRequest` not needed?

Comment (bapeey) [reply src/es/plottwistnofansub/src/eu/kanade/tachiyomi/extension/es/plottwistnofansub/PlotTwistNoFansub.kt]: I tested it without the headers and it works normally

**Pull request merged**